src/language-markdown/utils.js

```diff
@@ -125,16 +119,8 @@ function splitText(text, options) {
     const lastNode = getLast(nodes);
     if (lastNode && lastNode.type === "word") {
       if (
-        (lastNode.kind === KIND_NON_CJK &&
-          node.kind === KIND_CJ_LETTER &&
-          !lastNode.hasTrailingPunctuation) ||
-        (lastNode.kind === KIND_CJ_LETTER &&
-          node.kind === KIND_NON_CJK &&
-          !node.hasLeadingPunctuation)
-      ) {
-        nodes.push({ type: "whitespace", value: " " });
-      } else if (
         !isBetween(KIND_NON_CJK, KIND_CJK_PUNCTUATION) &&
+        !isBetween(KIND_CJK_PUNCTUATION, KIND_NON_CJK) &&
         // disallow leading/trailing full-width whitespace
         ![lastNode.value, node.value].some((value) => /\u3000/.test(value))
```

Comment:
In Simplified Chinese, 2 full-width spaces are used.

```latex
\documentclass[UTF8,fontset=ubuntu]{ctexart}
\usepackage[T1]{fontenc}
\begin{document}

\section{测试开始}

一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落一个段落

另一个段落另一个段落另一个段落另一个段落另一个段落另一个段落另一个段落另一个段落另一个段落另一个段落

最后一个段落最后一个段落最后一个段落最后一个段落最后一个段落最后一个段落最后一个段落最后一个段落最后一个段落
\end{document}
```

↓ Compile with XeLaTeX (in Overleaf)

![image](https://user-images.githubusercontent.com/12870451/139509191-e80682b3-72ba-48ed-9073-cd2a0c8a6ebe.png)

Reply:
I know that stable version also removes that. And I didn't blame you. I just want know why we did that. This question also not only asking you but other maintainers.

Reply:
@fisker 
I was relieved to hear that. It is likely to cause misunderstanding. I would have felt relieved if you had added something like "It is not your fault, though."

FYI:
- Introduced to `src/common/util.js` in #3026
- Moved from `src/common/util.js` to `src/language-markdown/utils.js` in #5040

As these PRs are locked, you might want to create a new issue on that.

If there are no other problems with this PR, I would be appreciated if you could approve and merge it.

P.S. You can reuse the following LaTeX source and preview for a new issue instead of the above LaTeX source for a Japanese document. (because I mixed Chinese words) You can also reuse the above LaTeX for Simplified Chinese documents.

```markdown
\documentclass[autodetect-engine]{bxjsarticle}

\begin{document}

\section{Let's begin the test!}

日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語

次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落次の段落

最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落最後の段落

\end{document}
```

↓Use the above `.latexmkrc` (or you can also use LuaLaTeX or XeLaTeX instead)

![image](https://user-images.githubusercontent.com/12870451/139536271-304fd77a-dfe5-4a0d-963b-ed16733ec085.png)